release_notes.txt

```diff
@@ -1,18 +1,48 @@
----------------------------------------------------------------------------------------------------
+################################################################################################
+Release Notes for OpenMDAO 2.1.0
+December 1, 2017
+
+New Features:
```

Comment:
Might be worth mentioning `RegularGridInterpComp`, which was moved from pycycle.

Reply:
And worth now changing it to the new name.